docs/developer-env.md

```diff
@@ -0,0 +1,534 @@
+- Parameter schemas
+- DAG previews
+
+See {ref}`vs-code-page` for more information about the Nextflow extension.
+
+````{tabs}
+
+```{group-tab} Windows
+
+To install the Nextflow VS Code extension on Windows:
+
+1. Open VS Code.
+1. Open the **Extensions** view in the left-hand menu.
+1. Search for **Nextflow**.
+1. Select **Install**.
+
+```
+
+```{group-tab} macOS
+
+To install the Nextflow VS Code extension on macOS:
+
+1. Open VS Code.
+1. Open the **Extensions** view in the left-hand menu.
+1. Search for **Nextflow**.
+1. Select **Install**.
+
+```
+
+```{group-tab} Linux
+
+To install the Nextflow VS Code extension on Linux Debian/Ubuntu distributions:
+
+1. Open VS Code.
+1. Open the **Extensions** view in the left-hand menu.
+1. Search for **Nextflow**.
+1. Select **Install**.
+
+```
+
+````
+
+
+
+### nf-core
+
+The [nf-core extensionpack](https://marketplace.visualstudio.com/items?itemName=nf-core.nf-core-extensionpack) adds a selection of tools that help develop with nf-core, a community effort to collect a curated set of analysis pipelines built using Nextflow.
+
+The nf-core extension pack includes 12 extensions:
+
+- [Apptainer/Singularity](https://marketplace.visualstudio.com/items?itemName=onnovalkering.vscode-singularity) - Provides syntax highlighting for Apptainer/Singularity definition files.
+- [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) - A basic spell checker that works well with camelCase code.
+- [Docker](https://marketplace.visualstudio.com/items?itemName=ms-azuretools.vscode-docker) - Makes it easy to create, manage, and debug containerized applications.
+- [EditorConfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) - Support for EditorConfig project files for code standardisation.
+- [indent-rainbow](https://marketplace.visualstudio.com/items?itemName=oderwat.indent-rainbow) - Highlight indentation levels, especially if they're inconsistent.
+- [Nextflow](https://marketplace.visualstudio.com/items?itemName=nextflow.nextflow) - Nextflow language support.
+- [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) - Code formatter using Prettier.
+- [Rainbow CSV](https://marketplace.visualstudio.com/items?itemName=mechatroner.rainbow-csv) - Highlight columns in .CSV files in different colors.
+- [Ruff](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff) - An extremely fast Python linter and code formatter, written in Rust.
+- [Todo Tree](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree) - Show TODO, FIXME, etc. comment tags in a tree view.
+- [YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) - YAML Language Support by Red Hat, with built-in Kubernetes syntax support.
+- [Markdown Extended](https://marketplace.visualstudio.com/items?itemName=jebbs.markdown-extended) - Gives Markdown previews, including admonitions.
```

Comment:
I'm slightly wary of actually listing them out like this, as it's the kind of thing that could get stale quite quickly. Would prefer something more generic like "a number of useful extensions that come recommended from community members, such as ..." then like 2 or 3 specific examples.

Reply:
I can see your point. I'll shrink this (and the other extensions) down

Reply:
I've shorted this text and removed the repeated code blocks.